base/broadcast.jl

```diff
@@ -40,15 +53,23 @@ promote_containertype(::Type{Array}, ct) = Array
 promote_containertype(ct, ::Type{Array}) = Array
 promote_containertype(::Type{Tuple}, ::Type{Any}) = Tuple
 promote_containertype(::Type{Any}, ::Type{Tuple}) = Tuple
+promote_containertype(::Type{Any}, ::Type{Nullable}) = Nullable
+promote_containertype(::Type{Nullable}, ::Type{Any}) = Nullable
+promote_containertype(::Type{Nullable}, ::Type{Array}) = Array{Nullable}
+promote_containertype(::Type{Array}, ::Type{Nullable}) = Array{Nullable}
+promote_containertype(::Type{Array{Nullable}}, ::Type{Array{Nullable}}) =
+    Array{Nullable}
+promote_containertype(::Type{Array{Nullable}}, ct) = Array{Nullable}
+promote_containertype(ct, ::Type{Array{Nullable}}) = Array{Nullable}
 promote_containertype{T}(::Type{T}, ::Type{T}) = T
 
 ## Calculate the broadcast indices of the arguments, or error if incompatible
 # array inputs
 broadcast_indices() = ()
 broadcast_indices(A) = broadcast_indices(containertype(A), A)
-broadcast_indices(::Type{Any}, A) = ()
+broadcast_indices(::Union{Type{Any}, Type{Nullable}}, A) = ()
```

Comment:
The type information is computed assuming `Nullable` is a scalar, but that information is not fully propagated through. Perhaps it should be?